app/javascript/schools/courses/components/students_editor/StudentsEditor__BulkImportForm.res

```diff
@@ -0,0 +1,354 @@
+
+  let formData =
+    ReactEvent.Form.target(event)->DomUtils.EventTarget.unsafeToElement->DomUtils.FormData.create
+
+  let url = "/school/courses/" ++ courseId ++ "/bulk_import_students"
+
+  Api.sendFormData(
+    url,
+    formData,
+    json => {
+      Json.Decode.field("success", Json.Decode.bool, json)
+        ? Notification.success(t("done_exclamation"), t("success_notification"))
+        : ()
+      send(EndSaving)
+    },
+    () => send(FailSaving),
+  )
+}
+
+let tableHeader = {
+  <thead>
+    <tr className="bg-gray-200">
+      <th className="text-left text-xs"> {"no" |> str} </th>
+      <th className="text-left text-xs"> {"name" |> str} </th>
+      <th className="text-left text-xs"> {"email" |> str} </th>
+      <th className="text-left text-xs"> {"title" |> str} </th>
+      <th className="text-left text-xs"> {"team_name" |> str} </th>
+      <th className="text-left text-xs"> {"tags" |> str} </th>
+      <th className="text-left text-xs"> {"affiliation" |> str} </th>
+    </tr>
+  </thead>
+}
+
+let csvDataTable = (csvData, fileInvalid) => {
+  ReactUtils.nullIf(
+    <table className="table-auto mt-5 border w-full overflow-x-scroll">
+      {tableHeader}
+      <tbody>
+        {csvData
+        |> Array.mapi((index, studentData) =>
+          <tr key={string_of_int(index)}>
+            <td className="border border-gray-400 truncate text-xs px-2 py-1">
+              {string_of_int(index + 2) |> str}
+            </td>
+            <td className="border border-gray-400 truncate text-xs px-2 py-1">
+              {StudentCSVData.name(studentData)->Belt.Option.getWithDefault("") |> str}
+            </td>
+            <td className="border border-gray-400 truncate text-xs px-2 py-1">
+              {StudentCSVData.email(studentData)->Belt.Option.getWithDefault("") |> str}
+            </td>
+            <td className="border border-gray-400 truncate text-xs px-2 py-1">
+              {StudentCSVData.title(studentData)->Belt.Option.getWithDefault("") |> str}
+            </td>
+            <td className="border border-gray-400 truncate text-xs px-2 py-1">
+              {StudentCSVData.teamName(studentData)->Belt.Option.getWithDefault("") |> str}
+            </td>
+            <td className="border border-gray-400 truncate text-xs px-2 py-1">
+              {StudentCSVData.tags(studentData)->Belt.Option.getWithDefault("") |> str}
+            </td>
+            <td className="border border-gray-400 truncate text-xs px-2 py-1">
+              {StudentCSVData.affiliation(studentData)->Belt.Option.getWithDefault("") |> str}
+            </td>
+          </tr>
+        )
+        |> React.array}
+      </tbody>
+    </table>,
+    fileInvalid->Belt.Option.isSome,
+  )
+}
+
+let clearFile = send => {
+  open Webapi.Dom
+
+  switch document |> Document.getElementById("csv-file-input") {
+  | Some(e) => HtmlInputElement.setValue(DomUtils.Element.unsafeToHtmlInputElement(e), "")
+  | None => ()
+  }
+
+  send(ClearCSVData)
+}
```

Comment:
@mahesh-krishnakumar Could you move this out to `DomUtils`, or somewhere more appropriate? This would be useful for #609.

Reply:
Done in https://github.com/pupilfirst/pupilfirst/commit/d85e08382631639a981a00e4f8c66d8ece52afb8